homeassistant/components/aussie_broadband/config_flow.py

```diff
@@ -0,0 +1,184 @@
+
+                self.data = user_input
+                self.services = await self.client.get_services()  # type: ignore[union-attr]
+
+                if not self.services:
+                    return self.async_abort(reason="no_services_found")
+
+                if len(self.services) == 1:
+                    return self.async_create_entry(
+                        title=self.data[CONF_USERNAME],
+                        data=self.data,
+                        options={CONF_SERVICES: [str(self.services[0][SERVICE_ID])]},
+                    )
+
+                # Account has more than one service, select service to add
+                return await self.async_step_service()
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_service(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the optional service selection step."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.data[CONF_USERNAME], data=self.data, options=user_input
+            )
+
+        service_options = {str(s[SERVICE_ID]): s["description"] for s in self.services}
+        return self.async_show_form(
+            step_id="service",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SERVICES, default=list(service_options.keys())
+                    ): cv.multi_select(service_options)
+                }
+            ),
+            errors=None,
+        )
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle reauth."""
+        errors: dict[str, str] | None = None
+        if user_input and user_input.get(CONF_USERNAME):
+            self._reauth_username = user_input[CONF_USERNAME]
+
+        elif self._reauth_username and user_input and user_input.get(CONF_PASSWORD):
+            data = {
+                CONF_USERNAME: self._reauth_username,
+                CONF_PASSWORD: user_input[CONF_PASSWORD],
+            }
+
+            if not (errors := await self.async_auth(data)):
+                entry = await self.async_set_unique_id(self._reauth_username.lower())
+                if entry:
+                    self.hass.config_entries.async_update_entry(
+                        entry,
+                        data=data,
+                    )
+                    await self.hass.config_entries.async_reload(entry.entry_id)
+                    return self.async_abort(reason="reauth_successful")
+                return self.async_create_entry(title=self._reauth_username, data=data)
+
+        return self.async_show_form(
```

Comment:
We need a separate confirm step to show the form for the reauth credentials.

Reply:
@MartinHjelmare do you have another integration I can reference? I know I had a lot of trouble getting reauth to work and referenced other implementations when creating this.
Just playing with it again now, it looks like the form gets submitted with the existing broken credentials which is why I implemented it the way I have.

Reply:
https://github.com/home-assistant/core/pull/65408/commits/aeb7b8b8c6ac6e550086c9a35bcc297d1cd9c6b8